rel/expr_darrow.go

```diff
@@ -15,7 +15,7 @@ type DArrowExpr struct {
 }
 
 // NewDArrowExpr returns a new DArrowExpr.
-func NewMapExpr(scanner parser.Scanner, lhs Expr, fn Expr) Expr {
+func NewDArrowExpr(scanner parser.Scanner, lhs Expr, fn Expr) Expr {
 	return &DArrowExpr{ExprScanner{scanner}, lhs, ExprAsFunction(fn)}
```

Comment:
seems fine, but I think `MapExpr` is a better name, what do you think @marcelocantos?

Reply:
Perhaps, but I would have thought at least something like SetMapExpr for consistency with the others.

Reply:
I chose `Map`, but I changed my mind because the term is too ambiguous.